lib/iris/analysis/maths.py

```diff
@@ -120,29 +120,25 @@ def _assert_is_cube(cube):
 def _assert_compatible(cube, other):
```

Comment:
Does this function even do anything? It's used in one place, right after a try/except block around 'da.broadcast_to'. So it seems like any broadcast errors would already have been thrown.